src/clr-angular/data/datagrid/datagrid-row.spec.ts

```diff
@@ -87,6 +87,54 @@ export default function(): void {
       });
     });
 
+    describe('Conditional selection', function() {
+      let context: TestContext<ClrDatagridRow, SelectableRow>;
+      let selectionProvider: Selection;
+
+      let radio: HTMLElement;
+      let checkbox: HTMLElement;
+
+      beforeEach(function() {
+        context = this.create(ClrDatagridRow, SelectableRow, DATAGRID_SPEC_PROVIDERS);
+        selectionProvider = TestBed.get(Selection);
+        TestBed.get(Items).all = [{ id: 1 }, { id: 2 }];
+      });
+
+      it('should not toggle when clrDgSelectable is false for type SelectionType.Single', function() {
+        selectionProvider.selectionType = SelectionType.Single;
+        context.testComponent.clrDgSelectable = false;
+        context.detectChanges();
+        radio = context.clarityElement.querySelector("input[type='radio']");
+
+        expect(radio.getAttribute('disabled')).toBe('true');
+        expect(radio.getAttribute('aria-disabled')).toBe('true');
+        context.clarityDirective.toggle();
+        expect(context.clarityDirective.selected).toBe(false);
+      });
+
+      it('should not toggle when clrDgSelectable is false for type  SelectionType.Multi', () => {
+        selectionProvider.selectionType = SelectionType.Multi;
+        context.testComponent.clrDgSelectable = false;
+        context.detectChanges();
+        checkbox = context.clarityElement.querySelector("input[type='checkbox']");
+
+        expect(checkbox.getAttribute('disabled')).toBe('true');
+        expect(checkbox.getAttribute('aria-disabled')).toBe('true');
+
+        context.clarityDirective.toggle();
+        expect(context.clarityDirective.selected).toBe(false);
+      });
+
+      it('should not be able you toggle the state of selection when clrDgSelectable is false', () => {
+        selectionProvider.selectionType = SelectionType.Multi;
+        context.clarityDirective.toggle(true);
+        context.testComponent.clrDgSelectable = false;
+        context.detectChanges();
+        context.clarityDirective.toggle();
+        expect(context.clarityDirective.selected).toBe(true);
```

Comment:
Looking at this again, this is one of the cases we want to allow. Users cannot call the `toggle()` method, they can only select the button (which is disabled). So we want to be able to allow the API to toggle a locked row but not a button click. That means line 149 of datagrid-row.ts also needs to be removed. This puts the API in agreement, where they can set the selected state manually through the input binding or use `toggle()`.

Reply:
Sorry for the late reply, had to test it before making any changes. It seems that it works as planned even without the toggle() return.